app/components/search/search.component.ts

```diff
@@ -1,11 +1,18 @@
 import { Component } from '@angular/core';
 
-
 @Component({
     selector:    'search',
     template: `
         <div class="searchContainer">
-            <input class="search" type="text" placeholder="search for courses" />
+            <input 
+                    class="search"
+                    #search
+                    type="text" 
+                    placeholder="search for courses"
+                    [value]="value"
+                    [(ngModel)]="value"
+                    (ngModelChange)="getInputValue($event)"
```

Comment:
You don't need `(ngModelChange)` output binding, if you already have 2-way data-binding `[(ngModel)]`. Output (`console.log`) model value on search icon click.